### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by dcxjn**

### CODE REVIEW

File: backend/src/authentication/auth.js [hunk 1]
```diff
@@ -1,18 +1,18 @@
 const jwt = require('jsonwebtoken')
-const User = require('../models/user')
+const UserAccount = require('../models/userAccount')
 
 const auth = async (req, res, next) => {
     try {
         const token = req.header('Authorization').replace('Bearer ', '')
         const decoded = jwt.verify(token, process.env.JWT_SECRET)
-        const user = await User.findOne({ _id: decoded._id, 'tokens.token': token })
+        const userAccount = await UserAccount.findOne({ _id: decoded._id, 'tokens.token': token })
```
Comment (aryans1204) [inline backend/src/authentication/auth.js]: @dcxjn you can leave the local variables as it is, just change the router file name to `userAccount.js` and the schema name to `UserAccount`

File: backend/src/authentication/auth.js [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 
-        if (!user) {
+        if (!userAccount) {
             throw new Error()
         }
 
         req.token = token
-        req.user = user
+        req.userAccount = userAccount
```
Comment (aryans1204) [inline backend/src/authentication/auth.js]: same thing here, let this be req.user coz otherwise there's a lot of refactoring to be done

**Review by aryans1204 — CHANGES_REQUESTED** [review]

**Comment by aryans1204** [comment]
yup, looks good now, @dcxjn anything else or can be merged?

**Comment by dcxjn** [comment]
> yup, looks good now, @dcxjn anything else or can be merged?

i have yet to change user to userAccount, but this can be merged first

**Comment by aryans1204** [comment]
> > yup, looks good now, @dcxjn anything else or can be merged?
> 
> i have yet to change user to userAccount, but this can be merged first

sure, however doing so just refactor the higher level modules like the schema name and the router names etc, and not the local variables. The reason for that is that a lot of local variables are used interchangeably as part of the request JSON, which if you only change in one place, will break the other endpoints (essentially what was happening before).

**Review by aryans1204 — APPROVED** [review]
LGTM 🚀

**Comment by aryans1204** [comment]
@dcxjn merging this. You can create a separate ticket for refactoring

**Pull request merged**